### PR TITLE
Implement Clock#javaClock

### LINKED ISSUE #6475: Clock should also provide a `java.time.Clock`
It would be very useful to be able to derive a `java.time.Clock` from a ZIO Clock, e.g. when integrating with Java Libraries that allow to manipulate time through a `java.time.Clock`. This allows us to use the `TestClock` even when Java Libraries are integrated in our project. 

Currently, I've managed to do this like this:

```scala
  def javaClockFromZIOClock(clock: Clock.Service, runtime: Runtime[_]): java.time.Clock = new java.time.Clock {
    override def getZone: java.time.ZoneId = java.time.ZoneId.systemDefault
    override def withZone(zone: java.time.ZoneId): java.time.Clock = this
    override def instant(): java.time.Instant = runtime.unsafeRun(clock.instant)
  }

 val myEffect = for {
  ... 
      clock <- ZIO.service[Clock.Service]
      runtime <- ZIO.runtime[Any]
      javaClock = javaClockFromZIOClock(clock, runtime)
```

However, for this to work I know need to have a `Runtime` in my environment and it feels kind of hacky.. is there another way to do this?

### PR DESCRIPTION
**Pull request by adamgfraser**
Resolves #6475.

### CODE REVIEW

File: core/shared/src/main/scala/zio/Clock.scala [hunk 1]
```diff
@@ -271,6 +275,20 @@ object Clock extends ClockPlatformSpecific with Serializable {
     def scheduler(implicit trace: ZTraceElement): UIO[Scheduler] =
       ZIO.succeed(globalScheduler)
 
+    def javaClock(implicit trace: ZTraceElement): UIO[java.time.Clock] = {
```
Comment (jdegoes) [inline core/shared/src/main/scala/zio/Clock.scala:278]: We may have a slight bias here for `asJava` or similar (`asJavaClock`?).

e.g.:

 - `asScheduledExecutorService`
 - `asDuration`

Comment (adamgfraser) [reply core/shared/src/main/scala/zio/Clock.scala:278]: Yes I am wondering if it should be `toJava` or `toJavaClock` since it returns an effect.

Comment (adamgfraser) [reply core/shared/src/main/scala/zio/Clock.scala:278]: It is a little weird when used as an accessor (`Clock.asJava` or `Clock.toJava`). Maybe a reason to leave as is. Or maybe we delete the accessor.

Comment (jdegoes) [reply core/shared/src/main/scala/zio/Clock.scala:278]: Good point! Maybe we leave as-is 👍

**Review by jdegoes — APPROVED** [review]

**Pull request merged**